Formula/libgccjit.rb

```diff
@@ -0,0 +1,146 @@
+    Dir["#{prefix}/**/*"].each do |f|
+      rm_rf f unless File.directory?(f) || File.basename(f).to_s.start_with?("libgccjit")
+    end
+  end
+
+  test do
+    (testpath/"test-libgccjit.c").write <<~EOS
+      #include <libgccjit.h>
+      #include <stdlib.h>
+      #include <stdio.h>
+
+      static void create_code (gcc_jit_context *ctxt) {
+          gcc_jit_type *void_type = gcc_jit_context_get_type (ctxt, GCC_JIT_TYPE_VOID);
+          gcc_jit_type *const_char_ptr_type = gcc_jit_context_get_type (ctxt, GCC_JIT_TYPE_CONST_CHAR_PTR);
+          gcc_jit_param *param_name = gcc_jit_context_new_param (ctxt, NULL, const_char_ptr_type, "name");
+          gcc_jit_function *func = gcc_jit_context_new_function (ctxt, NULL, GCC_JIT_FUNCTION_EXPORTED,
+                  void_type, "greet", 1, &param_name, 0);
+          gcc_jit_param *param_format = gcc_jit_context_new_param (ctxt, NULL, const_char_ptr_type, "format");
+          gcc_jit_function *printf_func = gcc_jit_context_new_function (ctxt, NULL, GCC_JIT_FUNCTION_IMPORTED,
+                  gcc_jit_context_get_type (ctxt, GCC_JIT_TYPE_INT), "printf", 1, &param_format, 1);
+          gcc_jit_rvalue *args[2];
+          args[0] = gcc_jit_context_new_string_literal (ctxt, "hello %s\\n");
+          args[1] = gcc_jit_param_as_rvalue (param_name);
+          gcc_jit_block *block = gcc_jit_function_new_block (func, NULL);
+          gcc_jit_block_add_eval (block, NULL, gcc_jit_context_new_call (ctxt, NULL, printf_func, 2, args));
+          gcc_jit_block_end_with_void_return (block, NULL);
+      }
+
+      int main (int argc, char **argv) {
+          gcc_jit_context *ctxt;
+          gcc_jit_result *result;
+          ctxt = gcc_jit_context_acquire ();
+          if (!ctxt) {
+              fprintf (stderr, "NULL ctxt\\n");
+              exit (1);
+          }
+          gcc_jit_context_set_bool_option (ctxt, GCC_JIT_BOOL_OPTION_DUMP_GENERATED_CODE, 0);
+          create_code (ctxt);
+          result = gcc_jit_context_compile (ctxt);
+          if (!result) {
+              fprintf (stderr, "NULL result\\n");
+              exit (1);
+          }
+          typedef void (*fn_type) (const char *);
+          fn_type greet = (fn_type)gcc_jit_result_get_code (result, "greet");
+          if (!greet) {
+              fprintf (stderr, "NULL greet\\n");
+              exit (1);
+          }
+          greet ("world");
+          fflush (stdout);
+          gcc_jit_context_release (ctxt);
+          gcc_jit_result_release (result);
+          return 0;
+      }
+    EOS
+    gcc = Formula["gcc"].bin/"gcc-10"
+    gcclib = Formula["gcc"].lib
+    tripple = `gcc -dumpmachine | grep -oE '[^.]+' | head -1`.strip
+    system gcc.to_s, "-I#{include}", "test-libgccjit.c", "-o", "test", "-L#{lib}", "-lgccjit", "-lgcc_ext.10.5"
+    lib_path = "LIBRARY_PATH=\"#{gcclib}/gcc/10/gcc/#{tripple}/10.2.0:/usr/local/Cellar/gcc/10.2.0/lib/gcc/10/\""
+    assert_equal "hello world\n", shell_output("#{lib_path} ./test")
```

Comment:
This might be harder to read, but it'll make sure we don't have to fix the test with every update
```suggestion
    gcc = Formula["gcc"].bin/"gcc-#{version.major}"
    tripple = `gcc -dumpmachine | grep -oE '[^.]+' | head -1`.strip
    system gcc.to_s, "-I#{include}", "test-libgccjit.c", "-o", "test", "-L#{lib}", "-lgccjit", "-lgcc_ext.10.5"
    lib_path = "LIBRARY_PATH=\"#{Formula["gcc"].lib}/gcc/#{version.major}/gcc/#{tripple}/#{version}:#{Formula["gcc"].opt_lib}/gcc/#{version.major}/\""
    assert_equal "hello world\n", shell_output("#{lib_path} ./test")
```